### ISSUE TITLE
Lazy loading of external provider openid/oauth client configuration data

### ISSUE THREAD

**Issue by danutzplusplus** [DuendeSoftware/Support#11]
Hello,

The [official docs mention how to configure the IdentityServer with external providers](https://docs.duendesoftware.com/identityserver/v5/ui/login/external/), but I was wondering if theere is any ability to allow for the openid/oauth client configuration data to not be provided during startup of the host, but to lazy load them after startup, but before actually needing them.

Currently these configuration values, passed via the OpenIdConnectOptions object, are only passed at startup. And I've not found any way to do this on demand, and in this way allowing for the values to be changed just before being used. And because of this the only way to change them is to perform a restart of the app, forcing them to be re-read.

So, to sum up, I would be great to know if there's any way or if anyone has any other idea to have some indirection when loading client configuration data from 

```
services.AddAuthentication()
    .AddOpenIdConnect("AAD", "Employee Login", options =>
    {
        options.SignInScheme = IdentityServerConstants.ExternalCookieAuthenticationScheme;

        // other options omitted
    });
```

to somewhere else, ideally in a on-demand/lazy manner/at the last possible moment. Or if there's any serious reason why this shouldn't be done.

I'd appreciate any feedback on this issue, even some alternatives as long as the changes are not significant to the way I'm hosting the app. Thank you.

**Comment by danutzplusplus** [comment]
Thanks, I had no idea it there was builtin support.
